src/analysis/convex/exposed.lean

```diff
@@ -0,0 +1,248 @@
+/-
+Copyright (c) 2021 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import analysis.convex.extreme
+
+/-!
+# Exposed sets
+
+This file defines exposed sets and exposed points for sets in a real vector space.
+
+An exposed set of `A` is a subset of `A` that maximizes a functional (a continuous linear map
+`E → ℝ`) over `A`. By convention, `∅` is exposed to all sets.
+This is an analytic notion of "being on the side of". It is stronger than being extreme (see
+`is_exposed.is_extreme`), but weaker (for exposed points) than being a vertex.
+
+An exposed set of `A` is sometimes called a "face of `A`", but we decided to reserve this
+terminology to the more specific notion of a face of a polytope (sometimes hopefully soon out
+on mathlib!).
+
+## Main declarations
+
+* `is_exposed A B`: States that `B` is exposed with respect to `A`.
+* `is_exposed.is_extreme`: An exposed set is also extreme.
+
+## References
+
+See chapter 8 of [*Barry Simon*, Convexity][simon2011]
+
+TODO:
+- define convex independence, intrinsic frontier and prove lemmas related to exposed sets and
+  points.
+- generalise to Locally Convex Topological Vector Spaces.
+More not-yet-PRed stuff is available on the branch `sperner_again`.
+-/
+
+open_locale classical affine big_operators
+open set
+
+variables {E : Type*} [normed_group E] [normed_space ℝ E] {x : E} {A B C : set E}
+  {X : finset E} {l : E →L[ℝ] ℝ}
+
+/-- A set is exposed with respect to `A` iff it maximizes some functional over `A`. -/
+def is_exposed (A B : set E) : Prop :=
+B.nonempty → ∃ l : E →L[ℝ] ℝ, B = {x ∈ A | ∀ y ∈ A, l y ≤ l x}
+
+
+/-- A useful way to build exposed sets from intersecting `A` with halfspaces. -/
+def continuous_linear_map.to_exposed (l : E →L[ℝ] ℝ) (A : set E) : set E :=
+{x ∈ A | ∀ y ∈ A, l y ≤ l x}
+
+lemma continuous_linear_map.to_exposed.is_exposed : is_exposed A (l.to_exposed A) := λ h, ⟨l, rfl⟩
+
+lemma is_exposed_empty : is_exposed A ∅ :=
+λ ⟨x, hx⟩, by { exfalso, exact hx }
+
+namespace is_exposed
+
+protected lemma subset (hAB : is_exposed A B) : B ⊆ A :=
+begin
+  rintro x hx,
+  obtain ⟨_, rfl⟩ := hAB ⟨x, hx⟩,
+  exact hx.1,
+end
+
+@[refl] lemma refl (A : set E) : is_exposed A A :=
+λ ⟨w, hw⟩, ⟨0, subset.antisymm (λ x hx, ⟨hx, λ y hy, by exact le_refl 0⟩) (λ x hx, hx.1)⟩
+
+lemma antisymm (hB : is_exposed A B) (hA : is_exposed B A) :
+  A = B :=
+hA.subset.antisymm hB.subset
+
+-- `is_exposed` is *not* transitive
```

Comment:
It's nice to add this but can you indicate why? Or sketch a simple counterexample